AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/ErrorMapping/KeychainStoreError+AuthConvertible.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSPluginsCore
+import Amplify
+
+extension KeychainStoreError: AuthErrorConvertible {
+
+    var authError: AuthError {
+        switch self {
+        case .configuration(let message):
+            return .configuration(message, self.recoverySuggestion)
+        case .unknown(let errorDescription, let error):
+            return .unknown(errorDescription, error)
+        case .conversionError(let errorDescription, let error):
+            return .configuration(errorDescription, self.recoverySuggestion, error)
+        case .codingError(let errorDescription, let error):
+            return .configuration(errorDescription, self.recoverySuggestion, error)
+        case .itemNotFound:
+            return .service(self.errorDescription, self.recoverySuggestion, self)
+        case .securityError:
+            return .service(self.errorDescription, self.recoverySuggestion)
```

Comment:
Is there a reason we're omitting the `underlyingError` for when mapping from `KeychainStoreError.securityError` to `AuthError.service`?

Reply:
We don't currently have an established pattern here, so I'm good either way. 

Just consider that if we add more information to `KeychainStoreError.securityError` or `KeychainStoreError.itemNotFound`, that information won't be propagated up through the `KeychainStoreError` --> `AuthError` transformation without updating this code also. If that's ok / desired, then omitting is fine.